checkout.go

```diff
@@ -83,6 +142,16 @@ func populateCheckoutOpts(ptr *C.git_checkout_options, opts *CheckoutOpts) *C.gi
 	ptr.disable_filters = cbool(opts.DisableFilters)
 	ptr.dir_mode = C.uint(opts.DirMode.Perm())
 	ptr.file_mode = C.uint(opts.FileMode.Perm())
+	ptr.notify_flags = C.uint(opts.NotifyFlags)
+	if opts.NotifyCallback != nil || opts.ProgressCallback != nil {
+		C._go_git_populate_checkout_cb(ptr)
+	}
+	if opts.NotifyCallback != nil {
+		ptr.notify_payload = pointerHandles.Track(*opts)
+	}
+	if opts.ProgressCallback != nil {
+		ptr.progress_payload = pointerHandles.Track(*opts)
+	}
```

Comment:
And aren't we asking `Track()` here to track the structure itself? We should be asking it to track the _pointer_ which would be `opts` itself.